### DISCUSSION TITLE
UART Voltage

### DISCUSSION THREAD

**Discussion by ToniPlays** [maland16/daly-bms-uart#23] [Q&A]
Hi!

I have done some of my own library programming based on this library, mainly due to different requirements. What I am currently wondering is if the bms UART logic voltage is 3.3v or 5.0v. I measured about 3.3v on the transmit pin but are the pins 5v tolerant?

Thanks

**Comment by maland16** [comment]
Ooh interesting. I measured 5V when I used my BMS with the PC adapter, so I imagine the pins are 5V tolerant but will work at 3.3V. It's also possible DALY changed from 5V to 3.3 since it's been 3 years since I bought mine, but I'm not sure. Did you measure 3.3 with the PC adapter or the bluetooth dongle?

**Reply by ToniPlays** [reply]
Hi, I measured about 3.6v on the TX pin which was kind of odd. Could also be due to my weird setup where the batteries are just a resistor array. I bought mine not too long ago (48V 100A li-ion). What model number does your have, mine is R25A1-IB06

Thanks

**Comment by avivsabilal29** [comment]
Hello, I have tried using the UART connection from the Daly BMS to UART with the RX and TX pins of the Daly BMS, and I have connected them to the TX and RX pins of the Arduino as well as the ground pin. In the first attempt, I succeeded, and the Arduino's serial monitor displayed the readings from the BMS. However, when I wanted to try it for the second time with the exact same circuit as the first attempt, the Arduino's serial monitor did not display any readings. Could this be because I have not connected a level shifter from the UART connection of the BMS to the Arduino?

**Comment by ResearchAssist01** [comment]
Hello, I am using Daly BMS 8-24S model R24TM, I have used Tx Rx and GND of UART to connect with ESP32. I just supplied power to the B- and B+ terminal and connected the NTC-A , but I received nothing from the BMS indicating the communication is not  happening between the ESP32 and Daly BMS. I have used the code provided by https://github.com/maland16/daly-bms-uart.git , what could be the issue ? @ToniPlays @avivsabilal29 @maland16 how you guys done it?

**Reply by avivsabilal29** [reply]
hello bro, I think it's the same issue when I tried. Maybe before deciding whether the bms is broken or not I prefer for you, first if you have a logic analyzer or oscilloscope try to check whether the data is really out or not. Secondly, you can also make sure whether the serial logic voltage is the same, namely 3.3v like the ESP32 pinout. This is important because if it is different, you need to add a level shifter to equalize the voltage. Third, another option is to try another MCU. I think there is a possibility that the serial connection has many errors.

**Reply by ResearchAssist01** [reply]
I’ve just unboxed a brand-new Daly BMS and connected it to my MCU Arduino UNO R4 . Unfortunately, I’m not seeing any TX activity from the MCU when I attempt to query the BMS. Here's what I've verified:
-Wiring: TX on MCU → RX on BMS, and RX back to MCU’s TX.
-Baud + settings: Serial1.begin(9600); in setup() and using 8‑N‑1.
-No data on logic analyzer when sending queries via bms.update() or getPackMeasurements().
Is there anything in the code or library that needs redefining to enable outgoing commands? I’m wondering if:
-A Serial1.begin() call is missing in the library or sketch for the BMS port.
-Specific initialisation (e.g. bms.Init(), or enabling host address or command bytes) is required before first send?

**Reply by gauix4** [reply]
I have the same problem no communication with an ESP32 S3 cable on a brand new BMS
